### ISSUE TITLE
Degree symbol \u00b0 won't parse

### ISSUE THREAD

**Issue by sn6uv** [mathics/Mathics#274]
This discussion continues on from that contained in #271

@wolfv At first sight that `GreekPi` example looks correct.

Probably the reason Mathics appeared not to start is that when you modify the parser it has to recreate `mathics/core/parsetab.py`. This takes ~15mins on my machine.

One thing I forgot to mention is that many of these named characters are included `mathics/core/characters.py`. Both Degree (l212) an Pi (l693) are included. In fact they even parse in Mathics:
```mathematica
In[1]:= \[Pi]
Out[1]= π

In[2]:= \[Degree]
Out[2]= °

In[3]:= \[Pi] == π
Out[3]= True

In[4]:= \[Degree] == °
Out[4]= True 
```

The behaviour in Mathematica is quite different:

```mathematica
In[1]:= 180 \[Degree]                                                           
Out[1]= 180 Degree

In[2]:= 180 \[Pi]                                                               
Out[2]= 180 Pi
```
I can't even seem to paste the unicode symbol into Mathematica's command line interface.

I think what happened is we added all these characters to the Mathics lexer as characters. E.g.
```mathematica
In[5]:= Head[\[Pi]]
Out[5]= Symbol
```
This was done to support unicode symbol names e.g.
```mathematica
In[6]:= 1 + \[Gamma]
Out[6]= 1 + γ

In[7]:= (\[Mu]x + 2) / \[Delta]
Out[7]= (2 + μx) / δ
```

It not immediately clear what the best way to fix this is.

**Comment by sn6uv** [comment]
@wolfv At first sight that `GreekPi` example looks correct.

Probably the reason Mathics appeared not to start is that when you modify the parser it has to recreate `mathics/core/parsetab.py`. This takes ~15mins on my machine.

One thing I forgot to mention is that many of these named characters are included `mathics/core/characters.py`. Both Degree (l212) an Pi (l693) are included. In fact they even parse

```mathematica
In[1]:= \[Pi]
Out[1]= π

In[2]:= \[Degree]
Out[2]= °
```

The behaviour in mathematica is quite different:

```mathematica
In[1]:= 180 \[Degree]                                                           
Out[1]= 180 Degree

In[2]:= 180 \[Pi]                                                               
Out[2]= 180 Pi
```
I can't even seem to paste the unicode symbol into Mathematica's command line interface.

**Comment by wolfv** [comment]
To me it seems that Mathematica is doing two additional parsing steps.

First, it goes from InputForm to FullForm, then calculates the results in that form, and then reconverts everything to OutputForm.

So basically one would like to go from an input like

3 ° -> Times[3, Degree] -> 3 °
3 Degree -> Times[3, Degree] -> 3 °

This seems to work fine for 'Dot[]' or any other operator. (ie. Dot[a,b] -> a . b). So maybe there should be some sort of constant parser in Mathics? That handles all the special cases (the ones on my mind: Pi, E, Degree) and matches the unicode and \[Degree] values?

If I type \[Degree] in the Mathematica GUI it is directly replaced with °. For some reason I was not able to test Mathematica without GUI as it didn't want to start with the --nogui option (weird..).

I'm just running a new test if Mathics is going to start after I added a rule ... But it really takes extremely long :) Let's see ...

**Comment by wolfv** [comment]
In the Mathematica GUI, this:

```
180 Degree
= 180 °
```

happens, btw. I think that's a point where the GUI and the terminal seem to differ?

**Comment by wolfv** [comment]
The output in the terminal seems to be the result of calling `TextForm[ ... ]`

I am unable to input any unicode in the Mathematica terminal interface as well, btw.

**Comment by sn6uv** [comment]
fixed by #441

**Comment by wolfv** [comment]
Awesome! :)
